report/filter.go

```diff
@@ -0,0 +1,170 @@
+		Endpoint:   "",
+		Status:     nil,
+		SLAGreater: 0,
+		SLALess:    100,
+		Message:    "",
+		PageNum:    1,
+		PageSize:   global.DefaultPageSize,
+		total:      0,
+		cnt:        0,
+	}
+}
+
+// Check check the filter is valid or not
+func (f *SLAFilter) Check() error {
+	log.Debugf("[Web] Check filter: %+v", f)
+	if f.SLAGreater > f.SLALess {
+		return fmt.Errorf("Error: Invalid SLA filter: gte(%0.2f) > (%0.2f)", f.SLAGreater, f.SLALess)
+	}
+	if f.SLAGreater > 100 || f.SLAGreater < 0 {
+		return fmt.Errorf("Error: Invalid SLA filter: gte(%0.2f), it must be between 0 - 100", f.SLAGreater)
+	}
+	if f.SLALess > 100 || f.SLALess < 0 {
+		return fmt.Errorf("Error: Invalid SLA filter: lte(%0.2f), it must be between 0 - 100", f.SLALess)
+	}
+	if f.PageNum < 1 {
+		return fmt.Errorf("Error: Invalid page number: %d, it must be greater than 0", f.PageNum)
+	}
+	if f.PageSize < 1 {
+		return fmt.Errorf("Error: Invalid page size: %d, it must be greater than 0", f.PageSize)
+	}
+	return nil
+}
+
+// HTML return the HTML format string
+func (f *SLAFilter) HTML() string {
+
+	span := "<span style=\"font-size:9pt; background-color:#666; color:white; padding:0 5px;border-radius: 3px;\">"
+	_span := "</span>  "
+
+	result := ""
+
+	if strings.TrimSpace(f.Name) != "" {
+		result += fmt.Sprintf(span+"<b>Name</b>: %s"+_span, f.Name)
+	}
+	if strings.TrimSpace(f.Kind) != "" {
+		result += fmt.Sprintf(span+"<b>Kind</b>: %s"+_span, f.Kind)
+	}
+	if strings.TrimSpace(f.Endpoint) != "" {
+		result += fmt.Sprintf(span+"<b>Endpoint</b>: %s"+_span, f.Endpoint)
+	}
+	if f.Status != nil {
+		result += fmt.Sprintf(span+"<b>Status</b>: %s"+_span, f.Status.String())
+	}
+	if strings.TrimSpace(f.Message) != "" {
+		result += fmt.Sprintf(span+"<b>Message</b>: %s"+_span, f.Message)
+	}
+	if f.SLAGreater > 0 || f.SLALess < 100 {
+		result += fmt.Sprintf(span+"<b>SLA</b>: %.2f%% - %.2f%% "+_span, f.SLAGreater, f.SLALess)
+	}
+
+	color := "#c00"
+	if f.PageNum <= f.cnt/f.PageSize+1 {
+		color = "#4E944F"
+	}
+	span = `<span style="font-size:9pt; background-color:` + color + `; color:white; padding:0 5px; margin-left:10px;border-radius: 3px;">`
+	result += fmt.Sprintf(span+"<b>Page %d / %d</b>"+_span, f.PageNum, f.cnt/f.PageSize+1)
+
+	span = `<span style="font-size:9pt; background-color:#4E944F; color:white; padding:0 5px; margin-left:10px;border-radius: 3px;">`
+	result += fmt.Sprintf(span+"<b>%d / %d Probers found!</b>"+_span, f.cnt, f.total)
+
+	result += "<br><br>"
+
+	return result
+}
+
+func (f *SLAFilter) getIndics() (start int, end int) {
+	start = (f.PageNum - 1) * f.PageSize
+	end = f.PageNum*f.PageSize - 1
+	return start, end
+}
+
+// Filter filter the probers
+func (f *SLAFilter) Filter(probers []probe.Prober) []probe.Prober {
+	start, end := f.getIndics()
+	f.cnt = 0
+	result := make([]probe.Prober, 0)
+	for _, p := range probers {
+		// if the name is not empty then filter by name
+		if strings.TrimSpace(f.Name) != "" && !strings.Contains(p.Name(), f.Name) {
```

Comment:
Trimming of properties of the Filter in the loop is inefficient,  it should be better to trim these fields out of the loop.